src/site/asciidoc/security.adoc

```diff
@@ -46,6 +46,89 @@ If you have encountered an unlisted security vulnerability or other unexpected b
+As such, it is the responsibility of the developer to properly demarcate sensitive data such that it can be consistently masked by log masking plugins.
+This sort of use case should make use of the `Message` API for better control over the output of such data.
+
+=== Availability
+
+Log4j goes to great lengths to minimize performance overhead along with options for minimizing latency or maximizing throughput.
+However, we cannot guarantee availability of the application if the appenders cannot keep up with the logs being written.
+Synchronous logging can cause applications to block and wait for a log message to be written.
+Asynchronous logging can also cause applications to block and wait depending on the wait strategy and queue full policy configured.
+Configuring too large or too many buffers in Log4j can also result in out of memory errors.
+
+=== Configuration Sources
+
+All configuration sources to an application must be trusted by the programmer.
+When loading a configuration file from disk (especially when a monitor interval is configured to reload the file periodically), the location of the configuration file must be kept safe from unauthorized modifications.
+Similarly, when loading a configuration file over the network such as through HTTP, this should be configured to use TLS or a secure connection in general with strong authentication guarantees.
+This remote location must be kept safe from unauthorized modifications.
+When configurations are modified through JMX, the JMX server should be safely configured to require authentication and a secure connection if being accessed over the network.
+When configurations are provided through JNDI, these should only use the `java` scheme for sharing configurations in a JavaEE or JakartaEE application service.
+JNDI-sourced configurations should not use other JNDI providers such as LDAP, DNS, or RMI, as all these providers are difficult to properly secure.
+
+=== Cryptographic Logs
+
+When using custom plugins to calculate message authentication codes, signatures, or encrypted logs, proper key management must be considered.
+Cryptography is a tricky subject, and key management lies at the heart of most design problems to solve when using it.
+If a custom plugin relies on a pre-shared key or password of some sort, it is likely broken beyond repair if being used for cryptographic purposes.
+Proper key management typically uses public key infrastructure (PKI) such as X.509 certificates, though this is not the only option.
+It is beyond the scope of Log4j to implement key management as this would depend on other infrastructure in use for validating or decrypting logs combined with the deployment and runtime environment of the logging application.
```

Comment:
Cryptographic logs are neither a feature we slightly offer, nor something we even briefly mention. This is not a place we educate users on cryptographic logs either. Hence... I think we can do without this section.

Reply:
True, it's not a current offering, but:

* The Flume plugin has some sort of half-assed support for encryption that violates these guidelines.
* We have a few feature requests related to this in Jira (mostly around encrypting logs, but there was another issue for signed logs).

I did research about this a few years ago, but my original implementation idea was fairly complex. Anyways, this section could perhaps be deferred until we have such a feature.